src/primitives/checksum.rs

```diff
@@ -200,3 +196,63 @@ macro_rules! impl_packed_fe32 {
 impl_packed_fe32!(u32);
 impl_packed_fe32!(u64);
 impl_packed_fe32!(u128);
+
+/// Iterator that yields the field elements that are input into a checksum algorithm for an [`Hrp`].
+pub struct HrpChecksumFe32Iter<'hrp> {
```

Comment:
In 5c9b549a3738cec82f633c14e274854b82edce45:

I'd drop `Checksum` from this name. I don't think it really clarifies anything.

Reply:
Nice, done.